current/learn-substrate/weight.md

```diff
@@ -19,6 +19,13 @@ used in mandatory inherent extrinsics) will generally be limited through economi
 simple terms, through transaction fees. The fee implications of the weight system are covered in the
 [Transaction Fees document](../runtime/fees).
 
+Substrates defines one unit of weight as one picosecond of execution time on some reference machine.
+Currently, that is an Intel Core i7-7700K CPU with 64GB of RAM and a NVMe SSD. Which reference
+machine is used ultimately depends on what are the hardware requirements the runtime developer
+wants to put on the validators. Weights are determined by benchmarking every dispatchable on this
+reference machine. Over time, the weight values will be adjusted to keep up with hardware
+and software improvements.
+
```

Comment:
I thought the reference hardware was supposed to be a universal standard. Having auniversal standard solves the problem of a pallet being written initially for one runtime, but then used in other runtimes. If different runtimes can have different reference hardware, then they will calculate weights differently. And we're straight back to pallets not being reusable in multiple runtimes.

My understanding was that if my chain wants to allow weaker hardware than the reference hardware you stated, then I should adjust my target block time and max block weight, not the definition of the reference hardware.

cc @shawntabrizi

Reply:
@JoshOrndorff you are correct here as far as I understand.

If your computer is 2x slower than our reference machine, you would set the `MaximumBlockWeight` to 1/2 the amount or you would double your block time.

In general though, FRAME makes an assumption about the hardware which defines our base units.

Reply:
Maybe I dont understand the misunderstanding :)

Reply:
Okay that makes sense. I will change this to clearly state that the default hardware is fixed and that developers should tweak the `MaximumBlockWeight`.

Reply:
Fixed.